woocommerce/wc-admin
# Cleanup items store actions

Fixes #5742

This PR makes the following changes to the "items" store:

* Eliminates errant product-as-action dispatch from `updateProductStock()`
* Fixes calls to `setError()`
* Add new `SET_ITEM` action for updating single items
* Add new `SET_ITEMS_TOTAL_COUNT` action to separate updating items and the total count of them
* Stop using `getItems()` inside `getItemsTotalCount` - the queries are different

### Detailed test instructions:

- Go to WooCommerce > Home
- Ensure you have an out of stock product
- Update the product stock
- Use Redux DevTools to verify there is not an action that is really the product object
- Smoke test reviews panel
- Smoke test other items endpoint consumers like category filter, product filter, 



### Changelog Note:



Dev: cleanup some items datastore code.
